packages/playback-core/src/autoplay.ts

```diff
@@ -43,23 +44,28 @@ export const setupAutoplay = (
 
   const updateHasPlayed = () => {
     // hasPlayed
-    mediaEl.addEventListener(
-      'playing',
-      () => {
-        hasPlayed = true;
-      },
-      { once: true }
-    );
+
+    if (!hasPlayed) {
+      addEventListenerWithTeardown(
+        mediaEl,
+        'playing',
+        () => () => {
```

Comment:
why does this have an extra arrow function wrapper? (Also, the other occurrences in this file)

Reply:
So the reason for this is I didn't want to only do half of the "Move fast, break things" mantra.